src/sentry/integrations/github/integration.py

```diff
@@ -256,7 +257,29 @@ def dispatch(self, request, pipeline):
             pipeline.bind_state("reinstall_id", request.GET["reinstall_id"])
 
         if "installation_id" in request.GET:
-            pipeline.bind_state("installation_id", request.GET["installation_id"])
-            return pipeline.next_step()
+            try:
+                # We want to limit GitHub integrations to 1 organization
+                installations_exist = OrganizationIntegration.objects.filter(
+                    integration=Integration.objects.get(external_id=request.GET["installation_id"])
+                ).exists()
+
+                if installations_exist:
+                    context = {
+                        "payload": {
+                            "success": False,
+                            "data": {
+                                "error": _("Github installed on another Sentry organization.")
+                            },
+                        }
+                    }
+                    return render_to_response(
+                        "sentry/integrations/github-integration-exists-on-another-org.html",
+                        context=context,
+                        request=request,
+                    )
+
+            except Integration.DoesNotExist:
```

Comment:
I think you want this to be an `else` block.

Reply:
@mgaeta hmm either the OrganizationIntegration record already exists or it doesn't and neither does the Integration record. This causes django to throw the Integration.DoesNotExist exception. If we don't capture it, it'll fail the pipeline